tools/generator/src/Generator+SetTargetConfigurations.swift

```diff
@@ -201,6 +201,34 @@ $(CONFIGURATION_BUILD_DIR)
                     .joined(separator: " ")
             }
 
+            if let swiftOutputs = target.outputs.swift {
+                let swiftmoduleOutputPaths = try swiftOutputs.paths(
+                    filePathResolver: filePathResolver
+                )
+                if !swiftmoduleOutputPaths.isEmpty {
+                    buildSettings["BAZEL_OUTPUTS_SWIFTMODULE"] =
+                        swiftmoduleOutputPaths.joined(separator: "\n")
+                }
+
+                if let generatedHeader = swiftOutputs.generatedHeader {
+                    buildSettings["BAZEL_OUTPUTS_SWIFT_GENERATED_HEADER"] =
+                        try filePathResolver.resolve(
+                            generatedHeader,
+                            useOriginalGeneratedFiles: true
+                        )
+                        .string
+                }
+            }
+
+            if let productOutput = target.outputs.product {
```

Comment:
Should we throw any errors in the case where some of these aren't set?

Looking at the run scripts, it seems like there might be some environment variables that are always expected to be set?

Reply:
What do you mean by when these are unset? They are optional because it will be valid for them to be unset for various types of targets.

Unless I'm misunderstanding what you are asking (probably).

Reply:
What I'm saying is that the scripts that check these variables (e.g .`BAZEL_OUTPUTS_PRODUCT`) do so unconditionally, so are there any cases where the script would try to run but the variable is not set? It's possible that the scripts are only configured to run in the generated projects under the same conditions that would cause these variables to be set, so they're guaranteed to be present when the script is run, but confirming that would require me to look at all the plumbing involved.

Finally, I'm slightly surprised to hear that some targets would produce no `outputs.product`, it seems like a target that produces nothing is unlikely, but I'm probably missing something.

Reply:
`outputs.product` is used for bundles and binaries. We don't set it for other outputs, because we don't need to copy the `.a` for example.

And right now, in this change, we only write out the script that uses these variables if they are set. #455 changes the script to check for the existence of the variables.